### ISSUE TITLE
Test Latest Dev on TN demo server

### ISSUE THREAD

**Issue by tdipisa** [geosolutions-it/ckanext-datitrentinoit#25]
We neeed to properly test what we recently developed providing a new deploy of all teh involved ckan's extensions into the geosolutions DEMO server 

http://opendata-trento.geo-solutions.it/

This means:

- [ ] upates for [ckanext-datitrentinoit](https://github.com/geosolutions-it/ckanext-datitrentinoit/commits/dcatapit)

- [ ] updates for [caknext-gsreport](https://github.com/geosolutions-it/ckanext-gsreport).

- [x] updates for [ckanext-extras](https://github.com/geosolutions-it/ckanext-extras)

- [x] installation of datapusher using the steps reported [here](https://demo.geo-solutions.it/share/opendatanetwork/doc/online/install/ckan/install_ext_data.html#datapusher) and check preview for semicolon delimited CSV resources

- [x]  check if modifications [reported here for #18](https://github.com/geosolutions-it/ckanext-datitrentinoit/issues/18#issuecomment-360549228) are correctly applied

- [x] check if the  configuration for custom licenses are correctly applied for #17

- [ ] Update the documentation below adding the installation steps for the new extensions (datapusher, extras, gsreport)

https://docs.google.com/document/d/1hqL4IqQ5yeEsso9YZYkoUNCnxSuDcZZsVumJmys3oNY/edit#heading=h.9sagfr48n50t

**Comment by cezio** [comment]
>  check if modifications reported here for #18 are correctly applied

```
[geosolutions@opendata-trento ckanext-spatial]$ git show
commit 44f05e181c4c70e3b062fec49e0f2251b52610a1
Author: amercader <amercadero@gmail.com>
Date:   Wed Aug 3 11:57:10 2016 +0100
```

```
[geosolutions@opendata-trento ckan]$ git branch
  master
* release-v2.5.2
[geosolutions@opendata-trento ckan]$ git diff --stat ckanext/reclineview/theme/public/vendor/recline/recline.js
 .../theme/public/vendor/recline/recline.js         |  156 ++++++++++----------
 1 files changed, 75 insertions(+), 81 deletions(-)
```

**Comment by tdipisa** [comment]
@cezio provide also a very quick doc describing what you did on the system

**Comment by cezio** [comment]
>  check if the configuration for custom licenses are correctly applied for #17

File is added in config, but i'm not sure if ckan was restarted after that change, because api returns different license set.

**Comment by cezio** [comment]
> @cezio provide also a very quick doc describing what you did on the system for the point above

I didn't do anything for that point, just checked if repos have correct commit/changes applied.

**Comment by cezio** [comment]
>  installation of datapusher using the steps reported here and check preview for semicolon delimited CSV resources

We are using python 2.7 for virtualenvs for ckan, while mod_wsgi (quite ancient version, 3.2) is compiled against py2.7, so it won't run properly datapusher properly:

```
==> /var/log/httpd/datapusher.error.log <==
'import site' failed; use -v for traceback
'import site' failed; use -v for traceback



[Wed May 30 15:58:12 2018] [error] [client 109.207.61.78] mod_wsgi (pid=17644): Target WSGI script '/usr/local/app_loc/geosolutions/conf/datapusher.wsgi' cannot be loaded as Python module.
[Wed May 30 15:58:12 2018] [error] [client 109.207.61.78] mod_wsgi (pid=17644): Exception occurred processing WSGI script '/usr/local/app_loc/geosolutions/conf/datapusher.wsgi'.
[Wed May 30 15:58:12 2018] [error] Traceback (most recent call last):
[Wed May 30 15:58:12 2018] [error]   File "/usr/local/app_loc/geosolutions/conf/datapusher.wsgi", line 3, in <module>
[Wed May 30 15:58:12 2018] [error]     import os
[Wed May 30 15:58:12 2018] [error] ImportError: No module named os
```

Until now datapusher was deployed as local http server with supervisor, available only locally (which is enough for local ckan). Is there any specific reason to expose it outside? Maybe we should reconsider leaving it as it was.

**Comment by cezio** [comment]
Few more details about mod_wsgi/datapusher:
 * mod_wsgi is from centos package and is compiled against py2.6
 * datapusher depends on sqlalchemy, which is installed with 1.2 version, which supports py2.7+
 * i tried to downgrade sqlalchemy to 1.1, which supports 2.6, but there are other packages (setuptools, python-dateutils) which needs to be at specific versions in order to install
 
 * other option is to compile mod_wsgi with py2.7 locally and use that instead of packaged module

**Comment by tdipisa** [comment]
@cezio that problem you reported depends by the CentOS distribution we are using un Trento VM.

(geosolutions) [geosolutions@opendata-trento local]$ ldd /etc/httpd/modules/mod_wsgi.so
        linux-vdso.so.1 =>  (0x00007ffe269d4000)
        **libpython2.6.so.1.0** => /usr/lib64/libpython2.6.so.1.0 (0x00007f1ae5694000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f1ae5477000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f1ae5272000)
        libutil.so.1 => /lib64/libutil.so.1 (0x00007f1ae506f000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f1ae4deb000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f1ae4a56000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f1ae5c69000)

this shows that is the mod_wsgi that referes the python 2.6 version directly.
So if the client does not upgrade the SO we need to rebuild the mod_wsgi for python 2.7 and use it instead of the current one.

Please do that documenting each step you did to provide this datapusher installation.

**Comment by cezio** [comment]
I've added Software Collections List repo (https://wiki.centos.org/AdditionalResources/Repositories/SCL) where mod_wsgi with support for py2.7 is present, but it want to update few packages as well (and bump apache/python versions a bit):

```
[root@opendata-trento ~]# yum install centos-release-scl
[root@opendata-trento ~]# yum info python27-mod_wsgi
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.mirrors.ovh.net
 * epel: mirror.freethought-internet.co.uk
 * extras: centos.mirrors.ovh.net
 * updates: mirrors.ircam.fr
Available Packages
Name        : python27-mod_wsgi
Arch        : x86_64
Version     : 4.5.13
Release     : 1.el6
Size        : 95 k
Repo        : centos-sclo-rh
Summary     : A WSGI interface for Python web applications in Apache
URL         : https://modwsgi.readthedocs.io/
License     : ASL 2.0
Description : The mod_wsgi adapter is an Apache module that provides a WSGI compliant
            : interface for hosting Python based web applications within Apache. The
            : adapter is written completely in C code against the Apache C runtime and
            : for hosting WSGI applications within Apache has a lower overhead than using
            : existing WSGI adapters for mod_python or CGI.

[root@opendata-trento ~]# yum install python27-mod_wsgi
Loaded plugins: fastestmirror
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: centos.mirrors.ovh.net
 * epel: mirror.freethought-internet.co.uk
 * extras: centos.mirrors.ovh.net
 * updates: mirrors.ircam.fr
Resolving Dependencies
--> Running transaction check
---> Package python27-mod_wsgi.x86_64 0:4.5.13-1.el6 will be installed
--> Processing Dependency: httpd24-httpd-mmn = 20120211x8664 for package: python27-mod_wsgi-4.5.13-1.el6.x86_64
--> Processing Dependency: libpython2.7.so.1.0()(64bit) for package: python27-mod_wsgi-4.5.13-1.el6.x86_64
--> Running transaction check
---> Package httpd24-httpd.x86_64 0:2.4.27-8.el6 will be installed
--> Processing Dependency: httpd24-httpd-tools = 2.4.27-8.el6 for package: httpd24-httpd-2.4.27-8.el6.x86_64
--> Processing Dependency: policycoreutils-python for package: httpd24-httpd-2.4.27-8.el6.x86_64
--> Processing Dependency: httpd24-runtime for package: httpd24-httpd-2.4.27-8.el6.x86_64
--> Processing Dependency: libnghttp2-httpd24.so.14()(64bit) for package: httpd24-httpd-2.4.27-8.el6.x86_64
--> Processing Dependency: libaprutil-1-httpd24.so.0()(64bit) for package: httpd24-httpd-2.4.27-8.el6.x86_64
--> Processing Dependency: libapr-1-httpd24.so.0()(64bit) for package: httpd24-httpd-2.4.27-8.el6.x86_64
---> Package python27-python-libs.x86_64 0:2.7.13-3.el6 will be installed
--> Processing Dependency: python27-python(abi) = 2.7 for package: python27-python-libs-2.7.13-3.el6.x86_64
--> Processing Dependency: python27-runtime for package: python27-python-libs-2.7.13-3.el6.x86_64
--> Running transaction check
---> Package httpd24-apr.x86_64 0:1.5.1-1.el6.1 will be installed
---> Package httpd24-apr-util.x86_64 0:1.5.4-1.el6 will be installed
---> Package httpd24-httpd-tools.x86_64 0:2.4.27-8.el6 will be installed
---> Package httpd24-libnghttp2.x86_64 0:1.7.1-1.el6 will be installed
---> Package httpd24-runtime.x86_64 0:1.1-18.el6 will be installed
--> Processing Dependency: scl-utils for package: httpd24-runtime-1.1-18.el6.x86_64
---> Package policycoreutils-python.x86_64 0:2.0.83-30.1.el6_8 will be installed
--> Processing Dependency: policycoreutils = 2.0.83-30.1.el6_8 for package: policycoreutils-python-2.0.83-30.1.el6_8.x86_64
--> Processing Dependency: libsemanage-python >= 2.0.43-4 for package: policycoreutils-python-2.0.83-30.1.el6_8.x86_64
--> Processing Dependency: audit-libs-python >= 1.4.2-1 for package: policycoreutils-python-2.0.83-30.1.el6_8.x86_64
--> Processing Dependency: setools-libs-python for package: policycoreutils-python-2.0.83-30.1.el6_8.x86_64
--> Processing Dependency: libselinux-python for package: policycoreutils-python-2.0.83-30.1.el6_8.x86_64
--> Processing Dependency: libcgroup for package: policycoreutils-python-2.0.83-30.1.el6_8.x86_64
---> Package python27-python.x86_64 0:2.7.13-3.el6 will be installed
--> Processing Dependency: python27-python-setuptools for package: python27-python-2.7.13-3.el6.x86_64
--> Processing Dependency: python27-python-pip for package: python27-python-2.7.13-3.el6.x86_64
---> Package python27-runtime.x86_64 0:1.1-25.el6 will be installed
--> Running transaction check
---> Package audit-libs-python.x86_64 0:2.4.5-6.el6 will be installed
--> Processing Dependency: audit-libs = 2.4.5-6.el6 for package: audit-libs-python-2.4.5-6.el6.x86_64
---> Package libcgroup.x86_64 0:0.40.rc1-24.el6_9 will be installed
---> Package libselinux-python.x86_64 0:2.0.94-7.el6 will be installed
---> Package libsemanage-python.x86_64 0:2.0.43-5.1.el6 will be installed
---> Package policycoreutils.x86_64 0:2.0.83-29.el6 will be updated
---> Package policycoreutils.x86_64 0:2.0.83-30.1.el6_8 will be an update
---> Package python27-python-pip.noarch 0:8.1.2-2.el6 will be installed
---> Package python27-python-setuptools.noarch 0:0.9.8-4.el6 will be installed
---> Package scl-utils.x86_64 0:20120927-29.el6_9 will be installed
---> Package setools-libs-python.x86_64 0:3.3.7-4.el6 will be installed
--> Processing Dependency: setools-libs = 3.3.7-4.el6 for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libsefs.so.4(VERS_4.0)(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libseaudit.so.4(VERS_4.2)(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libseaudit.so.4(VERS_4.1)(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libqpol.so.1(VERS_1.4)(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libqpol.so.1(VERS_1.3)(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libqpol.so.1(VERS_1.2)(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libpoldiff.so.1(VERS_1.3)(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libpoldiff.so.1(VERS_1.2)(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libapol.so.4(VERS_4.1)(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libapol.so.4(VERS_4.0)(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libsefs.so.4()(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libseaudit.so.4()(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libqpol.so.1()(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libpoldiff.so.1()(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Processing Dependency: libapol.so.4()(64bit) for package: setools-libs-python-3.3.7-4.el6.x86_64
--> Running transaction check
---> Package audit-libs.x86_64 0:2.4.5-3.el6 will be updated
--> Processing Dependency: audit-libs = 2.4.5-3.el6 for package: audit-2.4.5-3.el6.x86_64
---> Package audit-libs.x86_64 0:2.4.5-6.el6 will be an update
---> Package setools-libs.x86_64 0:3.3.7-4.el6 will be installed
--> Running transaction check
---> Package audit.x86_64 0:2.4.5-3.el6 will be updated
---> Package audit.x86_64 0:2.4.5-6.el6 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

=========================================================================================================================================
 Package                                   Arch                  Version                             Repository                     Size
=========================================================================================================================================
Installing:
 python27-mod_wsgi                         x86_64                4.5.13-1.el6                        centos-sclo-rh                 95 k
Installing for dependencies:
 audit-libs-python                         x86_64                2.4.5-6.el6                         base                           64 k
 httpd24-apr                               x86_64                1.5.1-1.el6.1                       centos-sclo-rh                103 k
 httpd24-apr-util                          x86_64                1.5.4-1.el6                         centos-sclo-rh                 88 k
 httpd24-httpd                             x86_64                2.4.27-8.el6                        centos-sclo-rh                1.3 M
 httpd24-httpd-tools                       x86_64                2.4.27-8.el6                        centos-sclo-rh                 81 k
 httpd24-libnghttp2                        x86_64                1.7.1-1.el6                         centos-sclo-rh                 56 k
 httpd24-runtime                           x86_64                1.1-18.el6                          centos-sclo-rh                1.0 M
 libcgroup                                 x86_64                0.40.rc1-24.el6_9                   updates                       131 k
 libselinux-python                         x86_64                2.0.94-7.el6                        base                          203 k
 libsemanage-python                        x86_64                2.0.43-5.1.el6                      base                           81 k
 policycoreutils-python                    x86_64                2.0.83-30.1.el6_8                   base                          437 k
 python27-python                           x86_64                2.7.13-3.el6                        centos-sclo-rh                 84 k
 python27-python-libs                      x86_64                2.7.13-3.el6                        centos-sclo-rh                5.8 M
 python27-python-pip                       noarch                8.1.2-2.el6                         centos-sclo-rh                1.7 M
 python27-python-setuptools                noarch                0.9.8-4.el6                         centos-sclo-rh                423 k
 python27-runtime                          x86_64                1.1-25.el6                          centos-sclo-rh                1.0 M
 scl-utils                                 x86_64                20120927-29.el6_9                   updates                        23 k
 setools-libs                              x86_64                3.3.7-4.el6                         base                          400 k
 setools-libs-python                       x86_64                3.3.7-4.el6                         base                          222 k
Updating for dependencies:
 audit                                     x86_64                2.4.5-6.el6                         base                          204 k
 audit-libs                                x86_64                2.4.5-6.el6                         base                           74 k
 policycoreutils                           x86_64                2.0.83-30.1.el6_8                   base                          663 k

Transaction Summary
=========================================================================================================================================
Install      20 Package(s)
Upgrade       3 Package(s)

Total download size: 14 M
Is this ok [y/N]:
```

**Comment by cezio** [comment]
There's also 3.x version (since we have mod_wsgi 3.2)
```
yum install python27-mod_wsgi-3.4-12.sc1.el6
....
=========================================================================================================================================
 Package                                   Arch                  Version                             Repository                     Size
=========================================================================================================================================
Installing:
 python27-mod_wsgi                         x86_64                3.4-12.sc1.el6                      centos-sclo-rh                 74 k
Installing for dependencies:
 audit-libs-python                         x86_64                2.4.5-6.el6                         base                           64 k
 httpd24-apr                               x86_64                1.5.1-1.el6.1                       centos-sclo-rh                103 k
 httpd24-apr-util                          x86_64                1.5.4-1.el6                         centos-sclo-rh                 88 k
 httpd24-httpd                             x86_64                2.4.27-8.el6                        centos-sclo-rh                1.3 M
 httpd24-httpd-tools                       x86_64                2.4.27-8.el6                        centos-sclo-rh                 81 k
 httpd24-libnghttp2                        x86_64                1.7.1-1.el6                         centos-sclo-rh                 56 k
 httpd24-runtime                           x86_64                1.1-18.el6                          centos-sclo-rh                1.0 M
 libcgroup                                 x86_64                0.40.rc1-24.el6_9                   updates                       131 k
 libselinux-python                         x86_64                2.0.94-7.el6                        base                          203 k
 libsemanage-python                        x86_64                2.0.43-5.1.el6                      base                           81 k
 policycoreutils-python                    x86_64                2.0.83-30.1.el6_8                   base                          437 k
 python27-python                           x86_64                2.7.13-3.el6                        centos-sclo-rh                 84 k
 python27-python-libs                      x86_64                2.7.13-3.el6                        centos-sclo-rh                5.8 M
 python27-python-pip                       noarch                8.1.2-2.el6                         centos-sclo-rh                1.7 M
 python27-python-setuptools                noarch                0.9.8-4.el6                         centos-sclo-rh                423 k
 python27-runtime                          x86_64                1.1-25.el6                          centos-sclo-rh                1.0 M
 scl-utils                                 x86_64                20120927-29.el6_9                   updates                        23 k
 setools-libs                              x86_64                3.3.7-4.el6                         base                          400 k
 setools-libs-python                       x86_64                3.3.7-4.el6                         base                          222 k
Updating for dependencies:
 audit                                     x86_64                2.4.5-6.el6                         base                          204 k
 audit-libs                                x86_64                2.4.5-6.el6                         base                           74 k
 policycoreutils                           x86_64                2.0.83-30.1.el6_8                   base                          663 k

Transaction Summary
=========================================================================================================================================
Install      20 Package(s)
Upgrade       3 Package(s)

Total download size: 14 M
Is this ok [y/N]
```

**Comment by tdipisa** [comment]
@cezio ok for the upgrades you mentioned if needed, but before doing that a shanpshot of the VM must be perfomed

**Comment by cezio** [comment]
As agreed on chat, I'm leaving datapusher deployed as it is now (supervisord + python http server). Deploying with mod_wsgi is not possible with current configuration and mod_wsgi  update from SCL repository in CentOS requires additional workarounds.

Documentation on datapusher deployment:

https://docs.google.com/document/d/1hqL4IqQ5yeEsso9YZYkoUNCnxSuDcZZsVumJmys3oNY/edit#heading=h.otxnn53a37ie

**Comment by cezio** [comment]
> installation of datapusher using the steps reported here and check preview for semicolon delimited CSV resources

This mostly works. There's bug in  ckanext-datapuser in ckan-2.5.2, which causes error 500 in logs:

![image](https://user-images.githubusercontent.com/116021/41367833-7bb4e316-6f40-11e8-8f82-5fe386ced0ca.png)

This is because parent dataset contains validation errors:

![image](https://user-images.githubusercontent.com/116021/41367875-965eb35e-6f40-11e8-9cac-d2c65bd34152.png)

This bug is fixed in ckan-2.5.9, meaning it will display dataset's error during upload to datapusher instead of error 500.

**Comment by cezio** [comment]
> upates for ckanext-datitrentinoit

Waiting for https://github.com/geosolutions-it/ckanext-datitrentinoit/pull/26

> updates for caknext-gsreport.

Waiting for https://github.com/geosolutions-it/ckanext-gsreport/pull/18
